### PR TITLE
Add toolbar sorting functionality

### PR DESCRIPTION
**Pull request by RMadjev**

### CODE REVIEW

**Comment by RMadjev** [comment]
Should we move this function in utils.js also?  The ASC/DSC button is WIP.

File: tcms/testplans/templates/testplans/get.html [hunk 1]
```diff
@@ -123,16 +123,16 @@ <h2 class="card-pf-title">
...
-                                                <ul class="dropdown-menu">
-                                                    <li class="selected"><a href="#">{% trans 'Name' %}</a></li>
-                                                    <li><a href="#">{% trans 'ID' %}</a></li>
-                                                    <li><a href="#">{% trans 'Status' %}</a></li>
-                                                    <li><a href="#">{% trans 'Priority' %}</a></li>
+                                                <ul class="dropdown-menu js-toolbar-sort-options">
+                                                    <li class="selected"><a>{% trans 'Summary' %}</a></li>
+                                                    <li><a>{% trans 'ID' %}</a></li>
+                                                    <li><a>{% trans 'Status' %}</a></li>
+                                                    <li><a>{% trans 'Priority' %}</a></li>
```
Comment (atodorov) [inline tcms/testplans/templates/testplans/get.html:135]: The list & the order should be the same as for the filtering drop down.

File: tcms/testplans/static/testplans/js/get.js [hunk 2]
```diff
@@ -244,6 +244,40 @@ function toolbarEvents() {
...
+        $('#sort-button')[0].innerHTML = sortBy + '<span class="caret"></span>';
+
+        sortBy = sortBy.toLowerCase();
+
+        let tcsParentElement = $('#testcases-list'),
+            visibleTCrows = $('.js-testcase-row:visible');
+
+        if (sortBy === 'status') {
+            sortBy = 'case_status';
+        }
```
Comment (atodorov) [inline tcms/testplans/static/testplans/js/get.js:261]: Unify both drop down lists (e.g. make them an includable HTML template) and use the data-filter-type attribute.

File: tcms/testplans/static/testplans/js/get.js [hunk 3]
```diff
@@ -244,6 +244,40 @@ function toolbarEvents() {
...
+
+        // reorder the tc rows
+        visibleTCrows.sort(function(tc1, tc2) {
+            let tc1Id = $(tc1).data('testcasePk'),
+                tc2Id = $(tc2).data('testcasePk');
+
+            if (sortBy.toLowerCase() === 'id') {
+                return tc2Id - tc1Id;
+            }
```
Comment (atodorov) [inline tcms/testplans/static/testplans/js/get.js:270]: This seems to sort strings, not numbers.

File: tcms/testplans/static/testplans/js/get.js [hunk 4]
```diff
@@ -244,6 +244,40 @@ function toolbarEvents() {
...
+
+            let value1 = allTestCases[tc1Id][sortBy],
+                value2 = allTestCases[tc2Id][sortBy];
```
Comment (atodorov) [inline tcms/testplans/static/testplans/js/get.js:273]: First I sort by ID then I sort by Status and I get `value1` is undefined.

File: tcms/testplans/templates/testplans/get.html [hunk 5]
```diff
@@ -324,6 +316,14 @@ <h2 class="card-pf-title">
     </div>
 </template>
 
+<template id="toolbar-dropdown">
+    <li class="selected" data-filter-type="summary"><a>{% trans 'Summary' %}</a></li>
+    <li data-filter-type="category"><a>{% trans 'Category' %}</a></li>
+    <li data-filter-type="priority"><a>{% trans 'Priority' %}</a></li>
+    <li data-filter-type="author"><a>{% trans 'Author' %}</a></li>
+    <li data-filter-type="default_tester"><a>{% trans 'Default tester' %}</a></li>
+</template>
```
Comment (atodorov) [inline tcms/testplans/templates/testplans/get.html:325]: FTR I meant an HTML file which you can `include` directly in this template but works like this for now. Will merge as-is.

**Review by atodorov — APPROVED** [review]

**Pull request merged**